### ISSUE TITLE
图像分割模型训练耗时

### ISSUE THREAD

**Issue by Jenny-Hard** [PaddlePaddle/Paddle#53266]
[问题描述：进行语义分割，训练数据102张1053*1053的图像，网络是UNet，用的aistudio32G显存的那组配置，模型训练大约需要多长时间呢，感谢赐教！
**目前运行近3个小时，还没有出第一个epoch的计算结果**

环境信息：
![studio环境信息](https://user-images.githubusercontent.com/30117827/233913787-55f0f2ea-c1a7-4fd7-b0ae-ce0e4916437e.PNG)

代码信息：
![code1](https://user-images.githubusercontent.com/30117827/233916146-195024b3-4444-4110-b99b-be151131906c.PNG)

![code2](https://user-images.githubusercontent.com/30117827/233916180-805627ad-b838-48ff-ba7d-a6a322840d8a.PNG)

![code3](https://user-images.githubusercontent.com/30117827/233916206-32c7361d-cdc5-4733-b4b9-703e27ec2112.PNG)

![code4](https://user-images.githubusercontent.com/30117827/233916264-ef32626c-04bf-49f6-9a47-ba220fafc078.PNG)

""" Parts of the U-Net model """
"""https://github.com/milesial/Pytorch-UNet/blob/master/unet/unet_parts.py"""

class DoubleConv(nn.Layer):
    """(convolution => [BN] => ReLU) * 2"""

    def __init__(self, in_channels, out_channels):
        super().__init__()
        self.double_conv = nn.Sequential(
            nn.Conv2D(in_channels, out_channels, kernel_size=3, padding=1),
            nn.BatchNorm2D(out_channels),
            nn.ReLU(),
            nn.Conv2D(out_channels, out_channels, kernel_size=3, padding=1),
            nn.BatchNorm2D(out_channels),
            nn.ReLU()
        )

    def forward(self, x):
        return self.double_conv(x)

class Down(nn.Layer):
    """Downscaling with maxpool then double conv"""

    def __init__(self, in_channels, out_channels):
        super().__init__()
        self.maxpool_conv = nn.Sequential(
            nn.MaxPool2D(2),
            DoubleConv(in_channels, out_channels)
        )

    def forward(self, x):
        return self.maxpool_conv(x)

class Up(nn.Layer):
    """Upscaling then double conv"""

    def __init__(self, in_channels, out_channels, bilinear=True):
        super().__init__()

        # if bilinear, use the normal convolutions to reduce the number of channels
        if bilinear:
            self.up = nn.Upsample(scale_factor=2, mode='bilinear', align_corners=True)
        else:
            self.up = nn.Conv2DTranspose(in_channels // 2, in_channels // 2, kernel_size=2, stride=2)

        self.conv = DoubleConv(in_channels, out_channels)

    def forward(self, x1, x2):
        # TODO: forward contains an Pad2D and Concat
        x = self.up(x2)
        h_diff = (x1.shape[2] - x.shape[2])
        w_diff = (x1.shape[3] - x.shape[3])
        pad = nn.Pad2D(padding = [h_diff//2,h_diff-h_diff//2,w_diff//2,w_diff-w_diff//2])
        x = pad(x)
        x = paddle.concat(x=[x1,x],axis=1)
        return self.conv(x)
"""
    def forward(self, x1, x2):
        x1 = self.up(x1)
        # input is CHW
        #diffY = paddle.to_tensor([x2.size()[2] - x1.size()[2]])
        #diffX = paddle.to_tensor([x2.size()[3] - x1.size()[3]])
        diffY = x2.shape[2] - x1.shape[2]
        diffX = x2.shape[3] - x1.shape[3]

        x1 = F.pad(x1, [diffX // 2, diffX - diffX // 2,
                        diffY // 2, diffY - diffY // 2])

        x = paddle.concat([x2, x1], dim=1)
        return self.conv(x)
"""

class OutConv(nn.Layer):
    def __init__(self, in_channels, out_channels):
        super(OutConv, self).__init__()
        self.conv = nn.Conv2D(in_channels, out_channels, kernel_size=1)

    def forward(self, x):
        return self.conv(x)

class UNet(nn.Layer):
    def __init__(self, n_channels, n_classes, bilinear=False):
        super(UNet, self).__init__()
        self.n_channels = n_channels
        self.n_classes = n_classes
        self.bilinear = bilinear

        self.inc = DoubleConv(n_channels, 64)
        self.down1 = Down(64, 128)
        self.down2 = Down(128, 256)
        self.down3 = Down(256, 512)
        self.down4 = Down(512, 512)
        self.up1 = Up(1024, 256, bilinear)
        self.up2 = Up(512, 128, bilinear)
        self.up3 = Up(256, 64, bilinear)
        self.up4 = Up(128, 64, bilinear)
        self.outc = OutConv(64, n_classes)

    def forward(self, x):
        x1 = self.inc(x)
        x2 = self.down1(x1)
        x3 = self.down2(x2)
        x4 = self.down3(x3)
        x5 = self.down4(x4)
        x = self.up1(x5, x4)
        x = self.up2(x, x3)
        x = self.up3(x, x2)
        x = self.up4(x, x1)
        logits = self.outc(x)
        return logits

![code5](https://user-images.githubusercontent.com/30117827/233916326-45a04968-dd88-4b37-8f56-8e306243a0b8.PNG)

![code6](https://user-images.githubusercontent.com/30117827/233916366-5f8381fb-fbd4-44dd-9e61-183f423bf38e.PNG)

**Comment by juncaipeng** [comment]
建议统计一下训练阶段每个步骤的耗时，同时确定是否是gpu上训练。
对于分割任务，推荐使用paddleseg套件进行开发。
